src/Agent.Plugins/TestFilePublisher/TestFilePublisher.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Agent.Plugins.Log.TestResultParser.Contracts;
+using Microsoft.TeamFoundation.TestClient.PublishTestResults;
+using Microsoft.VisualStudio.Services.WebApi;
+using ITestResultParser = Microsoft.TeamFoundation.TestClient.PublishTestResults.ITestResultParser;
+using ITestRunPublisher = Microsoft.TeamFoundation.TestClient.PublishTestResults.ITestRunPublisher;
+
+namespace Agent.Plugins.Log.TestFilePublisher
+{
+    public interface ITestFilePublisher
+    {
+        Task InitializeAsync();
+        Task PublishAsync();
+    }
+
+    public class TestFilePublisher : ITestFilePublisher
+    {
+        private readonly VssConnection _vssConnection;
+        private readonly PipelineConfig _pipelineConfig;
+        private readonly TraceListener _traceListener;
+        private readonly ITraceLogger _logger;
+        private readonly ITelemetryDataCollector _telemetry;
+        private ITestFileFinder _testFileFinder;
+        private ITestResultParser _testResultParser;
+        private ITestRunPublisher _testRunPublisher;
+
+        public TestFilePublisher(VssConnection vssConnection, PipelineConfig pipelineConfig, TraceListener traceListener,
+            ITraceLogger logger, ITelemetryDataCollector telemetry)
+        {
+            _traceListener = traceListener;
+            _vssConnection = vssConnection;
+            _pipelineConfig = pipelineConfig;
+            _logger = logger;
+            _telemetry = telemetry;
+        }
+
+        public TestFilePublisher(VssConnection vssConnection, PipelineConfig pipelineConfig, TraceListener traceListener,
+            ITraceLogger logger, ITelemetryDataCollector telemetry, ITestFileFinder testFileFinder,
+            ITestResultParser testResultParser, ITestRunPublisher testRunPublisher)
+        : this(vssConnection, pipelineConfig, traceListener, logger, telemetry)
+        {
+            _testFileFinder = testFileFinder;
+            _testResultParser = testResultParser;
+            _testRunPublisher = testRunPublisher;
+        }
+
+        public async Task InitializeAsync()
+        {
+            await Task.Run(() => Initialize());
+        }
+
+        public async Task PublishAsync()
+        {
+            var testResultFiles = new List<string>();
+            IList<TestRunData> testData;
+
+            var testRunContext = new TestRunContextBuilder("Auto Published Test Run")
+                .WithBuildId(_pipelineConfig.BuildId)
+                .WithBuildUri(_pipelineConfig.BuildUri)
+                .Build();
+
+            using (new SimpleTimer(TelemetryConstants.FindTestFilesAsync, _logger, TimeSpan.FromSeconds(60),
```

Comment:
>TimeSpan.FromSeconds(60) [](start = 83, length = 24)

the simple timer implementation today is not logging anything even if alotted time exceeds, see if you want to change that.